multitenancy-spring-boot-autoconfigure/src/test/java/org/axonframework/extensions/multitenancy/autoconfig/MultiTenancyAxonServerAutoConfigurationTest.java

```diff
@@ -0,0 +1,117 @@
+import org.axonframework.springboot.autoconfig.XStreamAutoConfiguration;
+import org.junit.jupiter.api.*;
+import org.junit.jupiter.api.extension.*;
+import org.springframework.boot.autoconfigure.AutoConfigurations;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.test.context.runner.ApplicationContextRunner;
+import org.springframework.context.annotation.EnableMBeanExport;
+import org.springframework.jmx.support.RegistrationPolicy;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * @author Stefan Dragisic
+ */
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration
+@EnableAutoConfiguration
+@EnableMBeanExport(registration = RegistrationPolicy.IGNORE_EXISTING)
+@TestPropertySource(properties = {"axon.axonserver.contexts=tenant-1,tenant-2"})
+class MultiTenancyAxonServerAutoConfigurationTest {
+
+    private final ApplicationContextRunner contextRunner = new ApplicationContextRunner()
+            .withConfiguration(
+                    AutoConfigurations.of(
+                            AxonAutoConfiguration.class,
+                            EventProcessingAutoConfiguration.class,
+                            InfraConfiguration.class,
+                            AxonServerBusAutoConfiguration.class,
+                            MultiTenancyAxonServerAutoConfiguration.class,
+                            AxonServerAutoConfiguration.class,
+                            NoOpTransactionAutoConfiguration.class,
+                            ObjectMapperAutoConfiguration.class,
+                            TransactionAutoConfiguration.class,
+                            XStreamAutoConfiguration.class,
+                            AxonTracingAutoConfiguration.class
+                    ));
+
+    @Test
+    void axonServerAutoConfiguration() {
+        contextRunner.withConfiguration(AutoConfigurations.of(MultiTenancyAxonServerAutoConfiguration.class))
+                     .withConfiguration(AutoConfigurations.of(MultiTenancyAutoConfiguration.class))
+                     .run(context -> {
+                         assertThat(context).getBean("tenantEventSchedulerSegmentFactory")
+                                            .isInstanceOf(TenantEventSchedulerSegmentFactory.class);
+                         assertThat(context).getBean("tenantAxonServerCommandSegmentFactory")
+                                            .isInstanceOf(TenantCommandSegmentFactory.class);
+                         assertThat(context).getBean("tenantAxonServerQuerySegmentFactory")
+                                            .isInstanceOf(TenantQuerySegmentFactory.class);
+                         assertThat(context).getBean("multiTenantQueryUpdateEmitter")
```

Comment:
If this test class is just about the factories, I don't think the query update emitter belongs here.